flasc/analysis/expected_power_analysis.py

```diff
@@ -0,0 +1,767 @@
+    use_standard_error: bool = True,
+    N: int = 1,
+    percentiles: List[float] = [2.5, 97.5],
+    remove_all_nulls_wd_ws: bool = False,
+    remove_any_null_turbine_bins: bool = False,
+    remove_any_null_turbine_std_bins: bool = False,
+    variance_only: bool = False,
+    fill_cov_with_var: bool = False,
+) -> ExpectedPowerAnalysisOutput:
+    """Calculate the total uplift in energy production using expected power methods.
+
+    Args:
+        a_in (AnalysisInput): An AnalysisInput object containing the dataframes to analyze
+        uplift_pairs (List[Tuple[str, str]]): A list of tuples containing the df_name
+            values to compare
+        uplift_names (List[str]): A list of names for the uplift results
+        test_turbines (List[int]): A list of turbine indices to test
+        wd_turbines (List[int]): A list of turbine indices for wind direction. Defaults to None.
+        ws_turbines (List[int]): A list of turbine indices for wind speed. Defaults to None.
+        use_predefined_wd (bool): Use predefined wind direction. Defaults to False.
+        use_predefined_ws (bool): Use predefined wind speed. Defaults to False.
+        wd_step (float): The step size for the wind direction bins. Defaults to 2.0.
+        wd_min (float): The minimum wind direction value. Defaults to 0.0.
+        wd_max (float): The maximum wind direction value. Defaults to 360.0.
+        ws_step (float): The step size for the wind speed bins. Defaults to 1.0.
+        ws_min (float): The minimum wind speed value. Defaults to 0.0.
+        ws_max (float): The maximum wind speed value. Defaults to 50.0.
+        bin_cols_in (List[str]): A list of column names to bin the dataframes by.
+            Defaults to ["wd_bin", "ws_bin"].
+        weight_by (str): The method to weight the bins. Defaults to "min".
+        df_freq (pd.DataFrame): A pandas dataframe with the frequency of each bin. Defaults to None.
+        use_standard_error (bool): Use standard error for the uplift calculation. Defaults to True.
+        N (int): The number of bootstrap samples. Defaults to 1.
+        percentiles (List[float]): The percentiles to calculate for the bootstrap samples.
+            Defaults to [2.5, 97.5].
+        remove_all_nulls_wd_ws (bool): Remove all null cases for wind direction and wind speed.
+             Defaults to False.
+        remove_any_null_turbine_bins (bool): When computing farm power, remove any bins where
+            and of the test turbines is null.  Defaults to False.
+        remove_any_null_turbine_std_bins (bool): Remove bins from consideration where any of
+            the covariance or num_points columns are null.  Defaults to False.
+        variance_only (bool): Only use the variance of turbine powers and not turbine-turbine
+            covariances. Defaults to False.
+        fill_cov_with_var (bool): Fill all missing covariance terms with the product of the
+            square root of the variances. Defaults to False.
```

Comment:
Reminder to raise an error if both `variance_only` and `fill_cov_with_var` are true. When both are true it's not really  clear which option gets used.

Reply:
Done!